src/domain/common/port/DataFetcherPort.ts

```diff
@@ -0,0 +1,3 @@
+export default interface DataFetcherPort {
+  fetch(url: string): Promise<string>;
```

Comment:
Is this an URL? I think this naming is the source of the confusion. In `Request` we have a `url` which is not exactly an URL but the value of a request, it describes more than just the URL. If we change this and maybe add more fields to the `Request` (`location`, `contentType`, `path`) then the other elements should come together nicely.

Reply:
No, it's a request. This name was changed before but it was then missed whilst merging in TS migration changes.

Yeah, I shall go with your `RequestExecutorFactory `suggestion and will consider enriching `Request` while doing it.